src/Sylius/Bundle/CoreBundle/Resources/config/services.xml

```diff
@@ -117,5 +117,15 @@
         </service>
 
         <service id="sylius.calculator.product_variant_price" class="Sylius\Component\Core\Calculator\ProductVariantPriceCalculator" />
+
+        <service id="sylius.translation_locale_provider.admin" class="Sylius\Component\Core\Provider\TranslationLocaleProvider">
+            <argument type="service" id="sylius.repository.locale" />
+            <argument type="string">%locale%</argument>
+        </service>
+
+        <service id="sylius.translation.translatable_listener.doctrine.orm" class="Sylius\Bundle\CoreBundle\EventListener\ORMTranslatableListener">
+            <argument type="service" id="service_container" />
+            <tag name="doctrine.event_subscriber" connection="default" priority="100" />
```

Comment:
This will be executed **before** ResourceBundle's `ORMTranslatableListener` (priority 99), so the current locale will never be the one retrieved from the context.